packages/pigeon/lib/objc_generator.dart

```diff
@@ -586,15 +587,20 @@ void _writeHostApiSource(Indent indent, ObjcOptions options, Api api) {
   assert(api.location == ApiLocation.host);
   final String apiName = _className(options.prefix, api.name);
 
-  void writeChannelAllocation(Method func, String varName) {
+  void writeChannelAllocation(Method func, String varName, String? taskQueue) {
     indent.writeln('FlutterBasicMessageChannel *$varName =');
     indent.inc();
-    indent.writeln('[FlutterBasicMessageChannel');
+    indent.writeln('[[FlutterBasicMessageChannel alloc]');
     indent.inc();
-    indent.writeln('messageChannelWithName:@"${makeChannelName(api, func)}"');
+    indent.writeln('initWithName:@"${makeChannelName(api, func)}"');
     indent.writeln('binaryMessenger:binaryMessenger');
-    indent
-        .writeln('codec:${_getCodecGetterName(options.prefix, api.name)}()];');
+    indent.write('codec:${_getCodecGetterName(options.prefix, api.name)}()');
+    if (taskQueue != null) {
+      indent.addln('');
+      indent.writeln('taskQueue:$taskQueue];');
+    } else {
+      indent.writeln('];');
+    }
```

Comment:
While you can't do the same thing here, why not make the `indent.writeln('];');` unconditional, and remove `];` from the version in the `if`? Not only would that avoid duplication now (in a way I think improves clarity significantly), it would scale to other optional parameters.